### ISSUE TITLE
MFA with wrong answer should return question/device response

### ISSUE THREAD

**Issue by javierjulio** [synapsepayments/synapse_pay-ruby#17]
When answering an MFA (question or device type) with the **wrong** answer the API responds with an empty APIList (Banks) object. When I output the contents of `banks.api_method` it shows that the response is the same as the `link` call where it returns the question. I take it the gem should be updated to handle this case no? To replicate in Sandbox mode use the following in a IRB console:

``` ruby
require 'synapse_pay'

# Set your SynapsePay credentials

client = SynapsePay::User.create({
  :email => "test-user123@testing.com",
  :fullname => "Test Account",
  :ip_address => "11.111.11.11",
  :phonenumber => "123456789"
})

mfa = client.banks.link({
  :username => "synapse_good",
  :password => "test1234",
  :bank => "Bank of America"
})

banks = mfa.answer("Bank of America", "wrong-test_answer")
```

The `banks` variable above will be an empty APIList instance:

``` sh
=> #<SynapsePay::APIList[SynapsePay::Bank]:0x3ff951cb6bc0> Data: [

]
```

But if you run the following you'll see that the `json` contents are the same response as the `link` call.

``` ruby
pp banks.api_method
```

prints out:

```
 @json=
  {:is_mfa=>true,
   :response=>
    {:access_token=>"...",
     :cookies=>"",
     :form_extra=>"",
     :mfa=>
      [{:question=>
         "I heard you like questions so we put a question in your question?"}],
     :type=>"questions"},
   :success=>true},
```

**Comment by joncalhoun** [comment]
I expected the server to return an error on this case, which is why we weren't looking to parse the response into anything but an error object if the answer was wrong. Let me run this by a few others to see what they think, but thanks for pointing it out.

**Comment by javierjulio** [comment]
Reading the blog post on MFA, in the [Mapping Resources section](http://api.synapsepay.com/v2.0/blog/banklogin-best-practices#mapping-responses) it suggests that another question can be returned so this definitely needs to be updated to handle different cases.

"... If another question is returned, follow the same steps."

**Comment by javierjulio** [comment]
Looks like we need some detail here @sankaet on what to expect. I've been meaning to ask but it seems that yes, it is possible to receive multiple questions but not in the same response but rather after answering the first question I might get a second question as part of the MFA process.

**Comment by joncalhoun** [comment]
This should be resolved in the latest version (0.0.8). Thanks for pointing it out!
